prog/dftb+/lib_dftbplus/initprogram.F90

```diff
@@ -1241,7 +1224,11 @@ subroutine initProgramVariables(this, input, env)
     ! Basic variables
     this%hamiltonianType = input%ctrl%hamiltonian
     this%tSccCalc = input%ctrl%tScc
-    this%tDFTBU = input%ctrl%tDFTBU
+    if (allocated(input%ctrl%dftbU_inp)) then
+      allocate(this%dftbU)
+      call TDftbU_init(this%dftbU, input%ctrl%dftbU_inp)
+      deallocate(input%ctrl%dftbU_inp)
```

Comment:
Let's leave away the `deallocate`, we are destroying input not much later anyway.

Reply:
OK, but there are various other input examples with deallocations, so shall we make this an issue for a future PR to clean up?

Reply:
Valid point, so probably it is indeed better to leave it this way for the moment.

Reply:
As its now changed, I'll leave it this way. But the question is whether to clean up these patterns globally in another PR if initprogram is getting refactored.